app/ExposureNotificationContext.tsx

```diff
@@ -31,38 +32,38 @@ interface ExposureNotificationProviderProps {
 const ExposureNotificationsProvider = ({
   children,
 }: ExposureNotificationProviderProps): JSX.Element => {
-  const [authorizationStatus, setAuthorizationStatus] = useState<DeviceStatus>(
-    initialStatus,
-  );
+  const [deviceStatus, setDeviceStatus] = useState<DeviceStatus>(initialStatus);
 
   useEffect(() => {
-    if (!isGPS) {
-      const subscription = BTNativeModule.subscribeToEnabledStatusEvents(
-        (status: DeviceStatus) => {
-          setAuthorizationStatus(status);
-        },
-      );
+    const subscription = BTNativeModule.subscribeToEnabledStatusEvents(
+      (status: DeviceStatus) => {
+        setDeviceStatus(status);
+      },
+    );
 
-      return () => {
-        subscription?.remove();
-      };
-    } else {
-      return () => {};
-    }
+    return () => {
+      subscription?.remove();
+    };
   }, []);
 
   const requestENAuthorization = () => {
-    const cb = (authorizationStatus: DeviceStatus) => {
-      setAuthorizationStatus(authorizationStatus);
+    const cb = (deviceStatus: DeviceStatus) => {
+      setDeviceStatus(deviceStatus);
     };
     BTNativeModule.requestAuthorization(cb);
   };
 
+  const getIsENAuthorizedAndEnabled = () => {
+    console.log(deviceStatus);
+    return deviceStatus[0] === 'AUTHORIZED' && deviceStatus[1] === 'ENABLED';
```

Comment:
i really wish js had pattern matching in elm you could say something like:
```elm
case deviceStatus of
  (Authorized, Enabled) -> true
  _ -> false
```

Reply:
Me too - I didn't love this piece either. If we come up with something better, I am all ears

Reply:
totally fine for now! but it might makes sense to wait to the last moment to make the decision and use a switch statement for the various code paths.  so we would skip the need for a function that goes from DeviceStatus to boolean altogether.

it also suggests to me that we might have the wrong data model here.  if from the UI perspective we only have the two states of `CanGetENData` and `CannotGetENData`, we might want to provide a type to represent this directly

```typescript
type CanCollectENData = boolean
type ENStatus = CanCollectENData
```

and provide a way to convert from the native data representation of `type DeviceStatus = [Authorization, Enablement]` to `ENStatus`

with something like:

```typescript
const toENStatus = (deviceStatus: DeviceStatus): ENStatus => {
  ...
```

though this might be getting to the touch pedantic side and im not sure its necessary.